### PR TITLE
Update/videopress add use token

### LINKED ISSUE #5488: Refactor sharable hooks in `edit`
There are a handful of behaviors in `[edit](https://github.com/Automattic/jetpack/blob/trunk/projects/packages/videopress/src/client/block-editor/blocks/video/edit.tsx)` that will be shared between the web and native versions. 
In some cases these behaviors include network requests which will require different handlers for the given platform.

### PR DESCRIPTION
**Pull request by jhnstn**
Fixes https://github.com/wordpress-mobile/gutenberg-mobile/issues/5488

## Proposed changes:

* Refactor edit.tsx to use a new custom hook to fetch the media token.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

N/A

## Does this pull request change what data or activity we track or use?

No

## Testing instructions:

TBD

### CODE REVIEW

File: projects/packages/videopress/src/client/block-editor/hooks/use-media-token.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+ * @returns {string} The media token.
+ */
+const useMediaToken = ( { guid, isPrivate, tokenType }: UseMediaTokenArgs ): string | null => {
+	const [ token, setToken ] = useState< string | null >( null );
+
+	useEffect( () => {
+		if ( ! isPrivate ) {
+			return setToken( null );
+		}
+		getMediaToken( tokenType, { id, guid } ).then( tokenData => {
```
Comment (fluiddot) [inline projects/packages/videopress/src/client/block-editor/hooks/use-media-token.ts:30]: Looks like the `id` parameter passed here is missing. In fact, I don't see it referenced in the previous code ([reference](https://github.com/Automattic/jetpack/pull/29024/files#diff-884251507b133a31140b452f625b28062a669ef32271f13f4fb832beb3a82460L151)). Maybe we could remove it, wdyt?

**Review by fluiddot — CHANGES_REQUESTED** [review]
Apart from the issue I shared in a comment, I tested the changes in the web version, and worked as expected. Once we address the issue, we could consider the PR approved from my side.

File: projects/packages/videopress/src/client/block-editor/hooks/use-media-token.ts (hunk 1 above)
Comment (jhnstn) [reply projects/packages/videopress/src/client/block-editor/hooks/use-media-token.ts:30]: Ah yeah, the id param was just recently removed. Which actually caused some merge conflicts with this branch. I thought I took out all the usages but missed this one.

**Comment by jhnstn** [comment]
https://github.com/Automattic/jetpack/pull/28945 removes the need for this PR

**Pull request closed without merging**